src/main/java/com/github/algobot76/surabaya/util/Analyzer.java

```diff
@@ -30,18 +40,29 @@ public Analyzer() {
 
 	public Project analyze(Resource file) {
 		FileUnzipper.unzip(file, destpath);
-		SourceRoot sourceRoot = new SourceRoot(
-				CodeGenerationUtils.mavenModuleRoot(SurabayaApplication.class).resolve("unzipped"));
-		try {
-			List<ParseResult<CompilationUnit>> results = sourceRoot.tryToParse();
-			for (ParseResult result : results) {
-				CompilationUnit cu = (CompilationUnit) result.getResult().get();
-				staticAnalysisVisitor.visit(cu, project);
-			}
+		JavaParser parser = new JavaParser();
+		Path path = CodeGenerationUtils.mavenModuleRoot(SurabayaApplication.class).resolve("unzipped");
+		try (Stream<Path> paths = Files.walk(path)) {
+			paths.forEach(p -> {
+				if (p.toString().endsWith(".java")) {
+					try {
+						String content = Files.readString(p, StandardCharsets.US_ASCII);
+						ParseResult result = parser.parse(content);
+						CompilationUnit cu = (CompilationUnit) result.getResult().get();
+						String[] lines = content.lines().toArray(String[]::new);
+						staticAnalysisVisitor.setFileSrc(lines);
+						staticAnalysisVisitor.visit(cu, project);
+					}
+					catch (IOException e) {
+						logger.error(e.getMessage());
+					}
+				}
+			});
```

Comment:
This is unnecessary as `SourceRoot` will handle the file reading properly for us.

Reply:
SourceRoot doesn't give access to original code, that's why I switched to JavaParsing each file alongside reading flie contents with Files.readString